pkg/api/types.go

```diff
@@ -276,6 +277,65 @@ type TestStepConfiguration struct {
 	// artifacts to upload. If unset, this will default under
 	// the repository root to _output/local/artifacts.
 	ArtifactDir string `json:"artifact_dir"`
+
+	// Only one of the following can be not-null.
+	ContainerTestConfiguration                      *ContainerTestConfiguration                      `json:"container,omitempty"`
+	OpenshiftAnsibleClusterTestConfiguration        *OpenshiftAnsibleClusterTestConfiguration        `json:"openshift_ansible,omitempty"`
+	OpenshiftAnsibleSrcClusterTestConfiguration     *OpenshiftAnsibleSrcClusterTestConfiguration     `json:"openshift_ansible_src,omitempty"`
+	OpenshiftInstallerClusterTestConfiguration      *OpenshiftInstallerClusterTestConfiguration      `json:"openshift_installer,omitempty"`
+	OpenshiftInstallerSmokeClusterTestConfiguration *OpenshiftInstallerSmokeClusterTestConfiguration `json:"openshift_installer_smoke,omitempty"`
+}
+
+// ContainerTestConfiguration describes a test that runs a
+// command in one of the previously built images.
+type ContainerTestConfiguration struct {
+	// From is the image stream tag in the pipeline to run this
+	// command in.
+	From PipelineImageStreamTagReference `json:"from"`
+}
+
+// ClusterType determines which cloud provider should be
+// used to provision the cluster.
+type ClusterType string
+
+// Types of clusters supported by the provisioners.
+const (
+	ClusterTypeAWS ClusterType = "aws"
+	ClusterTypeGCP             = "gcp"
+)
+
+// ClusterTestConfiguration describes a test that provisions
+// a cluster and runs a command in it.
+type ClusterTestConfiguration struct {
+	ClusterType ClusterType `json:"cluster_type"`
+}
+
+// OpenshiftAnsibleClusterTestConfiguration describes a test
+// that provisions a cluster using openshift-ansible and runs
+// conformance tests.
+type OpenshiftAnsibleClusterTestConfiguration struct {
```

Comment:
Do we not want the upgrade scenarios in here, too?

Reply:
I think I mentioned this earlier today: I left the upgrade and all the azure templates out for now because they change every time I look. We can always add them later.

Reply:
ACK